File: listfragment/src/main/java/com/adazhdw/listfragment/layout/InterceptLayoutEx.java
package com.adazhdw.listfragment.layout;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.widget.FrameLayout;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class InterceptLayoutEx extends FrameLayout {
    private String TAG = "InterceptFrameLayout";
    public InterceptLayoutEx(@NonNull Context context) {
        super(context);
    }

    public InterceptLayoutEx(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public InterceptLayoutEx(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent event) {
        if (mOnSlipActionListener!=null) {
            mOnSlipActionListener.OnSlipAction(event);
        }
        return super.onInterceptTouchEvent(event);
    }


    private OnMotionEventListener mOnSlipActionListener;
    public void setOnMotionEventListener(OnMotionEventListener listener){
        this.mOnSlipActionListener = listener;
    }
    public interface OnMotionEventListener {
        void OnSlipAction(MotionEvent event);
    }
}

File: list-fragment/src/main/java/com/adazhdw/list/layout/InterceptLayoutEx.kt
package com.adazhdw.list.layout

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.widget.FrameLayout

class InterceptLayoutEx : FrameLayout {
    private val TAG = "InterceptFrameLayout"


    private var mOnSlipActionListener: OnMotionEventListener? = null

    constructor(context: Context) : super(context) {}

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {}

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    override fun onInterceptTouchEvent(event: MotionEvent): Boolean {
        if (mOnSlipActionListener != null) {
            mOnSlipActionListener!!.OnSlipAction(event)
        }
        return super.onInterceptTouchEvent(event)
    }

    fun setOnMotionEventListener(listener: OnMotionEventListener) {
        this.mOnSlipActionListener = listener
    }

    interface OnMotionEventListener {
        fun OnSlipAction(event: MotionEvent)
    }
}
